python/cuml/cuml/tests/test_spectral_embedding.py

```diff
@@ -0,0 +1,137 @@
+    fetch_openml,
+    load_digits,
+    make_s_curve,
+    make_swiss_roll,
+)
+from sklearn.manifold import SpectralEmbedding as skSpectralEmbedding
+from sklearn.manifold import trustworthiness
+from sklearn.model_selection import train_test_split
+
+from cuml.manifold import SpectralEmbedding, spectral_embedding
+
+# Test parameters
+N_NEIGHBORS = 15
+N_COMPONENTS = 2
+
+
+def generate_s_curve(n_samples):
+    """Generate S-curve dataset."""
+    X, _ = make_s_curve(n_samples=n_samples, noise=0.05, random_state=42)
+    return X
+
+
+def generate_swiss_roll(n_samples):
+    """Generate Swiss Roll dataset."""
+    X, _ = make_swiss_roll(n_samples=n_samples, noise=0.1, random_state=42)
+    return X
+
+
+def load_mnist(n_samples):
+    """Load and sample MNIST dataset."""
+    mnist = fetch_openml("mnist_784", version=1, as_frame=False, parser="auto")
+    X, y = mnist.data, mnist.target.astype(np.int32)
+    X = X / 255.0
+    X, _, _, _ = train_test_split(
+        X, y, train_size=n_samples, stratify=y, random_state=42
+    )
+    return X
+
+
+def load_digits_dataset(n_samples=None):
+    """Load digits dataset (n_samples is ignored as dataset has fixed size)."""
+    digits = load_digits()
+    return digits.data
+
+
+# Dataset configurations: (dataset_loader, dataset_name, n_samples, min_trustworthiness)
+dataset_configs = [
+    (generate_s_curve, "S-curve", 1500, 0.8),
+    (generate_s_curve, "S-curve", 2000, 0.8),
+    (generate_swiss_roll, "Swiss Roll", 2000, 0.8),
+    (generate_swiss_roll, "Swiss Roll", 3000, 0.8),
+    (load_mnist, "MNIST", 5000, 0.8),
+    (load_digits_dataset, "Digits dataset", None, 0.8),
+]
+
+
+@pytest.mark.parametrize(
+    "dataset_loader,dataset_name,n_samples,min_trustworthiness",
+    dataset_configs,
+)
+def test_spectral_embedding_trustworthiness(
+    dataset_loader, dataset_name, n_samples, min_trustworthiness
+):
+    """Test trustworthiness comparison between sklearn and cuML on various datasets."""
+    # Load/generate dataset
+    X = dataset_loader(n_samples) if n_samples else dataset_loader(None)
+
+    # sklearn embedding
+    sk_spectral = skSpectralEmbedding(
+        n_components=N_COMPONENTS,
+        n_neighbors=N_NEIGHBORS,
+        affinity="nearest_neighbors",
+        random_state=42,
+        n_jobs=-1,
+    )
+    X_sklearn = sk_spectral.fit_transform(X)
+
+    # cuML embedding
+    X_gpu = cp.asarray(X)
+    cuml_spectral = SpectralEmbedding(
+        n_components=N_COMPONENTS, n_neighbors=N_NEIGHBORS, random_state=42
+    )
+    X_cuml_gpu = cuml_spectral.fit_transform(X_gpu)
+    X_cuml = cp.asnumpy(X_cuml_gpu)
+
+    # Calculate trustworthiness scores
+    trust_sklearn = trustworthiness(X, X_sklearn, n_neighbors=N_NEIGHBORS)
+    trust_cuml = trustworthiness(X, X_cuml, n_neighbors=N_NEIGHBORS)
+
+    # Assertions
+    assert trust_sklearn > min_trustworthiness
+    assert trust_cuml > min_trustworthiness
+
+
+def test_spectral_embedding_function_api():
```

Comment:
Would you mind adding a test for both the `spectral_embedding` function and the `SpectralEmbedding` class that verifies output type is plumbed through properly?

The best way to do this is to pass in a type and assert the output type is as expected. Something like (untested, and you'll need to fill in the `...`):

```python
def test_output_type_handling():
    X, _ = make_s_curve(...)  # X is a numpy array
    
    embedding = spectral_embedding(X, ...)
    assert isinstance(embedding, np.ndarray)
    
    # Test fit and attribute
    model = SpectralEmbedding(...).fit(X)
    assert isinstance(model.embedding_, np.ndarray)
    
    # Test fit_transform
    out = SpectralEmbedding(...).fit_transform(X)
    assert isinstance(out, np.ndarray)
```